### ISSUE TITLE
dismissAllModals promise needs to include closing animation time.

### ISSUE THREAD

**Issue by haipham-aula** [wix/react-native-navigation#5552]
### Issue Description

The event of closing modal is finished (promise resolved), but the closing animation is still in progress. Consequently, the second modal is actually rendered (with `componentDidMount` and stuff), but the view is invisible.

Related to #3733 and #5045

### Steps to Reproduce / Code Snippets / Screenshots

```js
// this won't work
await Navigation.dismissAllModals();
await Navigation.showModal(); // component is rendered, but invisible

// this works
await Navigation.dismissAllModals();
await wait(1000); // 1 sec
await Navigation.showModal(); // component is rendered, and visible
```

---
### Environment
* React Native Navigation version: 3.1.2
* React Native version: 0.60.5
* Platform(s) (iOS, Android, or both?): iOS
* Device info (Simulator/Device? OS version? Debug/Release?): iOS 13

**Comment by dcalhoun** [comment]
This is definitely still an issue.

**Comment by dcalhoun** [comment]
I believe this issue is still valid.

**Comment by danilobuerger** [comment]
Fixed by #6671
